src/mixins/object_animation.mixin.ts

```diff
@@ -78,58 +88,67 @@ export abstract class AnimatableObject<
    * @param {String} to Value to animate to
    * @param {Object} [options] Options object
    */
-  _animate<T>(key: string, to: T, options: TAnimationOptions = {}) {
+  _animate<T extends number | TColorArg>(
+    key: string,
+    to: T,
+    options: Partial<TAnimationOptions<T>> = {}
+  ) {
     const path = key.split('.');
     const propIsColor = this.colorProperties.includes(path[path.length - 1]);
     const currentValue = path.reduce((deep: any, key) => deep[key], this);
 
-    to = to.toString();
-    if (!propIsColor) {
-      if (~to.indexOf('=')) {
-        to = currentValue + parseFloat(to.replace('=', ''));
-      } else {
-        to = parseFloat(to);
-      }
+    if (!propIsColor && typeof to === 'string') {
+      to = to.includes('=')
+        ? currentValue + parseFloat(to.replace('=', ''))
```

Comment:
My understanding is that this logic is fine. But it *really* needs to be documented better, because it took my way too long to understand how this worked. It's fine though because the `+=50` syntax is for animating a property (specifically not a color) `50` units (pixels, I think) in whatever relevant direction. The tests show stuff like `.animate('left', '+=50')` which is supposed to shift the object 50 pixels to the left. So add 50 to the current left, as the code says.

`byValue` is more more about how much a value should be incremented every tick if I understand correctly. In the `Animation` constructor, the default of `byValue` is `endValue - startValue` which would make our animation 1 tick.

Reply:
Please double and check and correct me if I'm wrong, though. I wrote comments and stuff with that understanding.

Reply:
> `byValue` is more more about how much a value should be incremented every tick if I understand correctly

not every tick, but for the whole animation. It doesn't make sense to increment by a fixed value each tick because the frames aren't guaranteed to flush in a fixed delta so you would get uncertain values.

Comments seem correct, but as I wrote we have `byValue` for that so I will remove this option once @asturur approves